Sources/NIOSSH/SshAgent/Protocol.swift

```diff
@@ -0,0 +1,168 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// SSH Agent protocol as described here:
+// https://datatracker.ietf.org/doc/html/draft-miller-ssh-agent-17
+// and using the wire representation described here:
+// https://datatracker.ietf.org/doc/html/rfc4251#section-5
+//
+
+import Foundation
+import NIOCore
+
+enum MessageNumber: UInt8, Sendable {
+    case requestIdentities = 11
+    case signRequest = 13
+    case addIdentity = 17
+    case removeIdentity = 18
+    case removeAllIdentities = 19
+    case addSmartcardKey = 20
+    case removeSmartcardKey = 21
+    case lock = 22
+    case unlock = 23
+    case addIdConstrained = 25
+    case addSmartcardKeyContrained = 26
+    case messageExtension = 27
+    case failure = 5
+    case success = 6
+    case identitiesAnswer = 12
+    case signResponse = 14
+    case extensionFailure = 28
+    case extensionResponse = 29
+}
+
+/// A request to an SSH Agent
+///
+/// This enum handles the encoding of requests to `ByteBuffer`s
+public enum SshAgentRequest: Sendable {
```

Comment:
It's worth considering if this can be a `struct` not an `enum`. Currently, `enum`s in public APIs are very challenging: see https://github.com/swiftlang/swift-evolution/blob/main/proposals/0487-extensible-enums.md for more.